### ISSUE TITLE
Add --compare flag for baseline comparison

### ISSUE THREAD

**Issue by jharris1679** [AnswerLayer/sniffbench#12]
## Overview

Add a `--compare` flag to `sniff interview` that compares new agent responses against existing baselines using an LLM judge.

## Behavior

```bash
# First run establishes baseline (human graded)
sniff interview --agent claude-code

# Second run compares to baseline
sniff interview --agent opencode --compare
```

## Implementation

* Load existing baselines from `.sniffbench/baselines.json`
* For each case with a baseline, run LLM judge to compare:
  * New answer vs baseline answer
  * Score relative to baseline (better/same/worse)
* Show comparison results inline
* Optionally save new run for future comparison

## Dependencies

* Requires LLM-judge evaluator (separate issue)

## Acceptance Criteria

- [ ] `--compare` flag runs comparison mode
- [ ] Shows side-by-side or diff view of answers
- [ ] LLM judge scores new answer relative to baseline
- [ ] Summary shows which cases improved/regressed
